### PR TITLE
[tests-only][full-ci] Added webdav spaces in apiTrashbin suite

### PR DESCRIPTION
**Pull request by grgprarup**
## Description
Added Spaces examples for features inside suite apiTrashbin
Files covered:
- [x] trashbinDelete.feature
- [x] trashbinFilesFolders.feature
- [x] trashbinFilesFoldersOc10Issue23151.feature
- [x] trashbinSharingToRoot.feature
- [x] trashbinSharingToShares.feature
- [x] trashbinSkip.feature
## Related Issue

- Part of https://github.com/owncloud/ocis/issues/3065

## How Has This Been Tested?

- test environment:
- CI
- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by amrita-shrestha — COMMENTED** [review]
LGTM

File: tests/TestHelpers/WebDavHelper.php [hunk 1]
```diff
@@ -46,6 +46,22 @@ class WebDavHelper {
...
+	 *
+	 * @param string|null $user
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public static function removeSpaceIdForUser(
+		?string $user
+	):void {
+		if (\array_key_exists($user, self::$spacesIdRef) && \array_key_exists("personal", self::$spacesIdRef[$user])) {
```
Comment (kiranparajuli589) [inline tests/TestHelpers/WebDavHelper.php:60]: no user with duplicate username is pushed this condition should be enough imo :thinking:
```suggestion
		if (\array_key_exists($user, self::$spacesIdRef)) {
```

File: tests/TestHelpers/WebDavHelper.php [hunk 2]
```diff
@@ -46,6 +46,22 @@ class WebDavHelper {
...
 
+	/**
+	 * clear space id reference for user
+	 *
+	 * @param string|null $user
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public static function removeSpaceIdForUser(
```
Comment (kiranparajuli589) [inline tests/TestHelpers/WebDavHelper.php:57]:
```suggestion
	public static function removeSpaceIdReferenceForUser(
```

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by phil-davis — APPROVED** [review]

**Pull request merged**